### PR TITLE
feat(InteractiveOverlay): update Item and add Button and Link children

### PR DESCRIPTION
**Pull request by tristandubbeld**
Check out the components in Storybook. Also changed implementations of the `InteractiveOverlay` in the different cards, so make sure to check those out too.

### CODE REVIEW

File: packages/app/src/app/pages/Dashboard/Components/shared/StyledCard.tsx [hunk 1]
```diff
@@ -12,9 +12,6 @@ export const StyledCard = styled(Card)<StyledCardProps>(({ dimmed }) => ({
   '&:hover': {
     backgroundColor: '#252525',
   },
-  '&:has(button:hover)': {
-    backgroundColor: '#1D1D1D',
-  },
```
Comment (tristandubbeld) [inline packages/app/src/app/pages/Dashboard/Components/shared/StyledCard.tsx:17]: This was messing with the `Card` hover when it only contained one button. If we want to "reset" the card background when a (icon) button is hovered, we need to rethink this.

File: packages/components/src/components/InteractiveOverlay/InteractiveOverlay.tsx [hunk 2]
```diff
@@ -40,32 +26,76 @@ const StyledItem = styled.a`
...
+      if (React.isValidElement(child)) {
+        // TODO: fix ts-ignore "Property 'className' does not exist on type 'unknown'."
+        // @ts-ignore
+        return React.cloneElement(child, { className: styledClassName });
+      }
+
+      return child;
+    });
+
+    if (augmentedChildren.length) {
```
Comment (alexnm) [inline packages/components/src/components/InteractiveOverlay/InteractiveOverlay.tsx:88]: nitpick: can you add > 0 for the check here? for sanity :D

File: packages/components/src/components/InteractiveOverlay/InteractiveOverlay.tsx [hunk 3]
```diff
@@ -88,7 +118,9 @@ const Wrapper = styled(
     className: styledClassName,
   }: WrapperProps) => {
     if (React.Children.count(children) > 1) {
-      throw new Error('The Wrapper component can only contain one child.');
+      throw new Error(
+        'The StyledOverlay component can only contain one child.'
```
Comment (alexnm) [inline packages/components/src/components/InteractiveOverlay/InteractiveOverlay.tsx:122]: What is StyledOverlay from the perspective of the user? (design system consumer). Should we reference the InteractiveOverlay component instead?

**Review by alexnm — APPROVED** [review]
left a few notes, but looks good

File: packages/components/src/components/InteractiveOverlay/InteractiveOverlay.tsx (hunk 3 above)
Comment (tristandubbeld) [reply packages/components/src/components/InteractiveOverlay/InteractiveOverlay.tsx:122]: Sharp, will update this!

**Pull request merged**